### DISCUSSION TITLE
Need Help Installing on Mac

### DISCUSSION THREAD

**Discussion by murphybp2** [hardisgroupcom/sfdx-hardis#409] [Q&A] [answered]
I'm trying to get the extension working on my Mac but am running into errors that I can't find a resolution for.  I followed the installation instructions on here (https://sfdx-hardis.cloudity.com/#installation) best I could.  When I tried running the command 
`npm install @salesforce/cli --global`
in Terminal, I got this error. 

`npm ERR! code EACCES
npm ERR! syscall mkdir
npm ERR! path /usr/local/lib/node_modules/@salesforce
npm ERR! errno -13
npm ERR! Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/@salesforce'
npm ERR!  [Error: EACCES: permission denied, mkdir '/usr/local/lib/node_modules/@salesforce'] {
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'mkdir',
npm ERR!   path: '/usr/local/lib/node_modules/@salesforce'
npm ERR! }
npm ERR! 
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR! 
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator.`

I tried chaning the file permissions on the folder but that didn't work.  But I already have SFDX installed so I thought I may not need this. 

So I tried using the Extension in VS Code, and tried using the "Create files export configuration" command.  But I get this message when I tried it. 

` ›   Warning: hardis org files export is not a sfdx command.
Did you mean force data tree export? [y/n]: 
 ›   Error: Run sfdx help for a list of available commands.`

I wasnt sure what the issue was, but I thought maybe I needed to update my CLI.  So I ran SFDX update and got a message that sfdx is going away and now I need to use sf.  And now I can't even run SFDX as it says it the file no longer exists.  

So I guess I have to install SF now.  Will this work with SF?

**Comment by murphybp2** [comment, marked as answer]
Well, I ended up completely uinstalling SFDX and starting over from scratch, and it's working now.  I also installed NPM using HomeBrew instead of from the website.  Not sure if that made the difference, but it's working now.

**Reply by nvuillam** [reply]
You did the right thing :)
Sfdx / sf Windows & Mac installers are a mess, everything goes much better when installs are made using npm :)

**Reply by murphybp2** [reply]
Also wanted to say this extension saved my butt.  We had a process break and I needed to export the attachments for a ton of records to send to a team, and when I used your file export option it was simple and easy and saved me a ton of time.  This thing is great.  I appreciate you building it.

**Reply by nvuillam** [reply]
@Murphybp2 many thanks for your kind feedback ❤️ 

If you want to support us, you can:

- Star the repository -> https://github.com/hardisgroupcom/sfdx-hardis

![image](https://github.com/hardisgroupcom/sfdx-hardis/assets/17500430/aa03e314-0c14-4db9-9c66-cc81f6a0b621)


- share your experience on LinkedIn & other networks :)

**Reply by Heartist3** [reply]
@Murphybp2 Can you please share the process on how you did it?

**Reply by murphybp2** [reply]
@Heartist3 I don't recall exactly what I did beyond what I mentioned in my answer.  I removed everything that was already installed on my computer and then followed the install steps.  I used Homebrew to install NPM instead of the installer from the site.

**Reply by nvuillam** [reply]
@Heartist3 the key is to remove the sfdx-cli and/or sf cli that has been installed with the Salesforce provided executable installer (so u can switch to the npm installer)

**Reply by nvuillam** [reply]
@Murphybp2 if sfdx-hardis helped you, don't hesitate to post your feedback about it on LinkedIn 😇

My profile is https://www.linkedin.com/in/nicolas-vuillamy?utm_source=share&utm_campaign=share_via&utm_content=profile&utm_medium=android_app 🤓

**Reply by Heartist3** [reply]
@nvuillam and @Murphybp2 Thank you both for the assistance provided. Will try it out
